src/actions.ml

```diff
@@ -331,47 +329,111 @@ let bench_text ~job_info () =
+  >>= function
+  | Ok id -> (
+    match results with
+    | Ok results -> (
+        (* Formatting heleprs *)
+        let details summary text =
+          f "<details>\n<summary>%s</summary>\n\n%s\n\n</details>" summary text
+        in
+        let code_wrap str = f "```\n%s\n```" str in
+        (* Document *)
+        (* TODO: any more data here? *)
+        let summary_table =
+          details "Bench Summary" @@ code_wrap results.summary_table
+        in
+        let slow_table =
+          details (f "Top %d slow downs" results.slow_number)
+          @@ code_wrap results.slow_table
+        in
+        let fast_table =
+          details (f "Top %d speed ups" results.fast_number)
+          @@ code_wrap results.fast_table
+        in
+        ["Bench results:"; summary_table; slow_table; fast_table]
+        |> String.concat ~sep:"\n"
+        |> fun message ->
+        GitHub_mutations.post_comment ~bot_info ~id ~message
+        >>= function
+        | Ok _ ->
+            Lwt.return_unit
+        | Error e ->
+            Lwt_io.printlf "Unable to post bench comment for pr #%d: %s" number
+              e )
+    | Error e ->
+        Lwt_io.printlf "Unable to fetch_results for bench for pr #%d: %s" number
+          e )
+  | Error e ->
+      Lwt_io.printlf "Unable to get_pull_request_id for bench for pr #%d: %s"
+        number e
+
+let update_bench_status ~bot_info job_info (gh_owner, gh_repo) ~external_id
+    ~number =
   let open Lwt.Syntax in
-  let* repo_id =
-    GitHub_queries.get_repository_id ~bot_info ~owner:gh_owner ~repo:gh_repo
-  in
-  match repo_id with
-  (* Not currently using pr id that was fetched *)
-  | Ok repo_id -> (
-      Lwt_io.printl "Pushing status check for bench job."
-      <&>
-      let url = f "https://gitlab.com/coq/coq/-/jobs/%d" job_info.build_id in
-      let summary = f "## GitLab Job URL:\n[GitLab Bench Job](%s)\n" url in
-      let state = job_info.build_status in
-      let context = "bench" in
-      ( match state with
-      | "success" ->
-          let* text = bench_text ~job_info () in
-          Lwt.return_ok
-            (COMPLETED, Some SUCCESS, "Bench completed successfully", text)
-      | "failed" ->
-          let* text = bench_text ~job_info () in
-          Lwt.return_ok
-            (COMPLETED, Some NEUTRAL, "Bench completed with failures", text)
-      | "running" ->
-          Lwt.return_ok (IN_PROGRESS, None, "Bench in progress", "")
-      | "cancelled" | "canceled" ->
-          Lwt.return_ok
-          (COMPLETED, Some CANCELLED, "Bench has been cancelled", "")
-      | "created" ->
-          Lwt.return_error ("Bench job has been created, ignoring info update.")
-      | _ ->
-          Lwt.return_error (f "Unknown state for bench job: %s" state) )
+  match number with
+  | None ->
+      Lwt_io.printlf "No PR number provided for bench summary so aborting."
+  | Some number -> (
+      GitHub_queries.get_repository_id ~bot_info ~owner:gh_owner ~repo:gh_repo
```

Comment:
Actually, in the cases where we need both the repository id and the PR id, instead of doing two requests, we could only do the latter and amend it to also return the repository id.